File: Web/krk-web/src/app/shared/response/pagination.response.ts
export interface PaginationResponse<T> {
  data: T[],
  pageNumber: number,
  pageSize: number,
  totalPages: number,
}

export function emptyPaginationResponse<T>(pageNumber: number, pageSize: number): PaginationResponse<T> {
  return {
    data: [],
    pageNumber: pageNumber,
    pageSize: pageSize,
    totalPages: 1
  }
}

File: Web/Shared/Models/PaginationResponse.cs
namespace KRK_Shared.Models;

public class PaginationResponse<T>
{
    public List<T> Data { get; set; } = [];
    public int PageNumber { get; set; } = default!;
    public int PageSize { get; set; } = default!;
    public long TotalPages { get; set; } = default!;
}